en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.md

```diff
@@ -67,15 +90,123 @@ The PM03D power module is supplied with the board.
 - The flight controller is powered via the CLIK-Mate cable to `POWER1` or `POWER2` port
 - The RPi CM4 is powered by the `USB C` (CM4 Slave) connection.
+  You can also use your own power supply to power the RPi CM4 baseboard.
 
 The image below shows the wiring in greater detail.
 
 ![Image showing writing from the PM03D power module to the baseboard](../../assets/companion_computer/holybro_pixhawk_rpi_cm4_baseboard/baseboard_wiring_guide.jpg)
 
+## Flashing the RPi CM4
+
+This section explains how you install your preferred Linux distro, such as "Raspberry Pi OS 64bit" onto the RPi EMCC.
+
+
+Notes:
+
+- If you are using PX4, you will need to use PX4 version 1.13.1 or newer for PX4 to recognize this baseboard.
+- The fan does not indicate if the RPi CM4 is powered/running or not.
+- The power module plugged into Power1/2 does not power the RPi part.
+  You can use the additional USB-C Cable from the PM03D power module to the CM4 Slave USB-C port.
+- The Micro-HDMI port is an output port.
+- RPi CM4 boards that do not have Wifi device will not connect automatically.
+  In this case you will need to plug it into a router or plug a compatible Wifi dongle into the CM4 Host ports.
+
+### Flash EMMC
+
+To flash a RPi image onto EMMC.
+
+1. Switch Dip-Switch to `RPI`.
+
+   ![](../../assets/companion_computer/holybro_pixhawk_rpi_cm4_baseboard/cm4_dip_switch.png)
+1. Connect computer to USB-C _CM4 Slave_ port used to power & flash the RPi.
+
+   ![](../../assets/companion_computer/holybro_pixhawk_rpi_cm4_baseboard/cm4_usbc_slave_port.png)
+
+1. Get `usbboot`, build it and run it.
+
+   ```
+   sudo apt install libusb-1.0-0-dev
+   git clone --depth=1 https://github.com/raspberrypi/usbboot,
+   cd usbboot
+   make
+   sudo ./rpiboot
+   ```
+
+1. You can now install your preferred Linux distro using The `rpi-imager`.
+   Make sure you add WiFi and SSH settings (hidden behind the gear/advanced symbol).
+
+   ```
+   sudo apt install rpi-imager
+   rpi-imager
+   ```
+
+1. Once done, unplugging USB-C CM4 Slave (this will unmount the volumes, and power off the CM4).
+1. Switch Dip-Switch back to `EMMC`.
+1. Power on CM4 by providing power to USB-C CM4 Slave port.
+1. To check if it’s booting/working you can either:
+   - Check there is HDMI output
+   - Connect via SSH (if set up in rpi-imager, and WiFi is available).
+
+### Configure PX4 to CM4 MAVLink Serial Connection
+
+:::note
+If you are using [Ethernet](#ethernet-connection) to connect the FC and RPi, this setup is not needed.
+:::
+
+The Pixhawk FC module is [internally connected to the RPi CM4](#rpi-cm4-fc-serial-connection) using `TELEM2` (`/dev/ttyS4`).
+The FC and RPi CM4 must both be configured to communicate over this port.
+
+#### FC Serial Port Setup
+
+The FC should be set up to connect to the `TELEM2` port correctly by default.
+If not, you can configure the port using the parameters as shown.
+
+To enable this MAVLink instance on the FC:
+
+1. Set the parameters:
```

Comment:
@vincentpoont2 How do you connect to the FC to QGC so you can set these parameters 
- is it a particular USB port you should connect to>
- After setting up everything else, can you just connect to it via the RaPi wifi? (i.e. is everything routed?)

Reply:
Connect via USB Type C port on the baseboard labeled as "FC"